canlink/writer/converttostring.go

```diff
@@ -0,0 +1,61 @@
+package writer
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"encoding/json"
+	"go.uber.org/zap"
+)
+
+// Converts timestamped frames into strings for file writing
+func convertToAscii(l *zap.Logger, timestampedFrame *TimestampedFrame) string {
```

Comment:
this is pretty repetitive. Can you wrap the error handling in a local function to make it easier to read?

```go
	var builder strings.Builder

	write := func(value string) {
		if _, err := builder.WriteString(value); err != nil {
			l.Error(err.Error())
		}
	}

	write(timestampedFrame.Time.Format(_messageTimeFormat))
	write(" " + strconv.FormatUint(uint64(timestampedFrame.Frame.ID), _decimal))
	write(" Rx")
	write(" " + strconv.FormatUint(uint64(timestampedFrame.Frame.Length), _decimal))

	for i := uint8(0); i < timestampedFrame.Frame.Length; i++ {
		write(" " + fmt.Sprintf("%02X", timestampedFrame.Frame.Data[i]))
	}

	return builder.String()
```

this is from gpt, so make sure it's all good before using